src/business-logic/services/test/throttled-rundown-service.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { ThrottledRundownService } from '../throttled-rundown-service'
+import { instance, mock, verify, when } from '@typestrong/ts-mockito'
+import { Rundown, RundownInterface } from '../../../model/entities/rundown'
+import { ThrottledRundownException } from '../../../model/exceptions/throttled-rundown-exception'
+
+describe(ThrottledRundownService.name, () => {
+  describe('when two operations are performed within the locked interval', () => {
+    beforeEach(() => jest.useFakeTimers())
+    afterEach(() => jest.useRealTimers())
+    describe('throws an error to second operation', () => {
+      it('throws error when two take nexts are performed with 0ms delay', async () => {
+        const mockRepo: ThrottledRundownService = mock<ThrottledRundownService>()
```

Comment:
You are mocking the testee type here. It would be nice with a more descriptive name and another type. I assume that this is the rundown service that should be passed in to the throttled rundown service. If that is the case, then I will suggest:
```typescript
const mockedRundownService: RundownService = mock<RundownService>()
```

And since you have the same mock setup for this in all tests, I would extract the mock setup out into a separate function.

Reply:
First is fixed, the second I simplify the test and in reality now we don't need any external mock setup because we make the setup very small, when we remove unnecessary values.